.github/workflows/flutter-branch.yml

```diff
@@ -49,7 +53,8 @@ jobs:
           working-directory: ${{ inputs.working-directory }}
       - run: flutter analyze
   test:
-    runs-on: [self-hosted]
+    runs-on:
+      ["${{ inputs.use-mac-runners-for-test && 'macOS' || 'self-hosted'  }}"]
```

Comment:
Self-hosted needs to be there either way.

` ["self-hosted"${{ inputs.use-mac-runners-for-test && ', "macOS" || ' ' }}]`

Reply:
You could also do

`[${{ inputs.use-mac-runners-for-test && ' "self-hosted", "macOS" ' || ' "self-hosted" ' }}]`